### PR TITLE
Fix retries

### PR DESCRIPTION
**Pull request by sHermanGriffiths**
# Author's Checklist

Before assigning reviewers, step through the following checklist. Check each item once it's completed. If an item is skipped, write an explanation below the item.

- [ ] **Construction:** Review the code diff and confirm there's no dead code, debug code, or unnecessary comments.
- [ ] **Requirements:** Review the requirements in the issue for this PR and confirm they're all implemented.
- [ ] **Architecture:** Confirm that the changes to the code are consistent with the existing architecture.
- [ ] **SOUP/OTS Software:** Any new dependencies have appropriate licenses and are documented or pinned in requirements files.
- [ ] **Unit Test:** Atomic unit tests have been added, if appropriate.
- [ ] **End-to-End Tests:** End-to-end tests have been extended or updated, if appropriate.
- [ ] **Low-level Documentation:** Comments or doc strings for affected code have been updated.
- [ ] **High-Level Documentation:** The README is updated, as appropriate.
- [ ] **Change Log:** New features or bug fixes have been added to the change log in the README.

### CODE REVIEW

File: n2y/notion.py [hunk 1]
```diff
@@ -320,6 +272,7 @@ def get_database_pages(self, database_id, filter=None, sorts=None):
         notion_pages = self.get_database_notion_pages(database_id, filter, sorts)
         return [self._wrap_notion_page(np) for np in notion_pages]
 
+    @retry_api_call(max_retries)
```
Comment (johndgiese) [inline n2y/notion.py:275]: Can you remove `max_retries` here and just rely on the default?

File: n2y/errors.py [hunk 2]
```diff
@@ -98,6 +98,9 @@ class APIErrorCode(str, Enum):
     This can occur when the time to respond to a request takes longer than 60 seconds,
     the maximum request timeout."""
 
+    GatewayTimeoutError = "gateway_timeout"
```
Comment (johndgiese) [inline n2y/errors.py:101]: Are there any other missing keys that we should add?

File: n2y/notion.py (hunk 1 above)
Comment (sHermanGriffiths) [reply n2y/notion.py:277]: that is the default, if you checkout main.py to see where this variable is originally created. It needs to be passed here to allow for changes based on the arguments passed to the main function.
```py
13    max_retries = DEFAULT_MAX_RETRIES
...
22    def main(raw_args, access_token):
...
45        global max_retries
46        max_retries = args.max_retries
```

Comment (sHermanGriffiths) [reply n2y/notion.py:277]: @johndgiese

Comment (johndgiese) [reply n2y/notion.py:277]: Sounds good, thank you!

File: n2y/notion.py [hunk 3]
```diff
@@ -199,6 +136,23 @@ def _override_notion_classes(self, notion_object, object_types, default_object_t
                     f'override the base class "{base_class.__name__}"',
                 )
 
+    def _organize_notion_classes(
```
Comment (johndgiese) [inline n2y/notion.py:139]: 👍 splitting out some of these methods!

File: n2y/errors.py (hunk 2 above)
Comment (sHermanGriffiths) [reply n2y/errors.py:101]: Yes, I just added them. Nice Catch!

**Review by johndgiese — APPROVED** [review]
Nice work!

**Pull request merged**